project/packages/dtos/src/testCases.ts

```diff
@@ -0,0 +1,41 @@
+import { z } from "zod";
+
+const testCasesItemSchema = z
+  .object({
+    input: z.any(),
+    output: z.any(),
+  })
+  .catchall(z.any());
+
+export const testCasesSchema = z.object({
+  id: z.string().uuid(),
+  question_id: z.string().uuid(),
+  cases: z
+    .array(testCasesItemSchema)
+    .min(1, { message: "At least one test case is required" }),
+  schema: z.object({
+    type: z.literal("object", {
```

Comment:
only change from me: changed this to 
`type: z.literal("object"...` 
instead of 
`type: z.string().refine((val) => val === "object", {...`

Reply:
i think this zod validation is good enough, no need to complicate it further 😭